File: servers/jax-rs/src/main/java/com/wordnik/api/SchemaApi.java
package com.wordnik.api;

import com.wordnik.swagger.annotations.*;

import com.wordnik.client.model.Object;
import java.util.List;
import com.wordnik.api.NotFoundException;

import javax.ws.rs.core.Response;
import javax.ws.rs.*;

@Path("/schema.json")
@Api(value = "/schema", description = "the schema API")
@Produces({"application/json"})
public class SchemaApi {
  @GET
  @Path("/")
  @ApiOperation(value = "Get model schemata for data objects returned by this API.", notes = "", responseClass = "Object")
  @ApiErrors(value = { @ApiError(code = 200, reason = "Request was successful"),@ApiError(code = 400, reason = "Parameter Error"),@ApiError(code = 401, reason = "Unauthorized"),@ApiError(code = 404, reason = "Not Found")})
     
  public Response find(
    @ApiParam(value = "Optional model filter. If omitted, will return all models."
    ,required=true
)@QueryParam("model")
 String model
    )
      throws NotFoundException {
      // do some magic!
      return Response.ok().entity(new ApiResponse(ApiResponse.OK, "magic!")).build();
  }

  @GET
  @Path("/websocketHelp")
  @ApiOperation(value = "Returns help text & subject list for websocket usage.", notes = "", responseClass = "Object")
  @ApiErrors(value = { @ApiError(code = 200, reason = "Request was successful"),@ApiError(code = 400, reason = "Parameter Error"),@ApiError(code = 401, reason = "Unauthorized"),@ApiError(code = 404, reason = "Not Found")})
     
  public Response websocketHelp(
    )
      throws NotFoundException {
      // do some magic!
      return Response.ok().entity(new ApiResponse(ApiResponse.OK, "magic!")).build();
  }

  }

File: servers/scalatra/src/main/scala/apis/SchemaApi.scala
package apis

import com.wordnik.client.model.Object
import java.io.File

import org.scalatra.{ TypedParamSupport, ScalatraServlet }
import org.scalatra.swagger._
import org.json4s._
import org.json4s.JsonDSL._
import org.scalatra.json.{ JValueResult, JacksonJsonSupport }
import org.scalatra.servlet.{FileUploadSupport, MultipartConfig, SizeConstraintExceededException}

import scala.collection.JavaConverters._

class SchemaApi (implicit val swagger: Swagger) extends ScalatraServlet 
    with FileUploadSupport
    with JacksonJsonSupport
    with SwaggerSupport {
  protected implicit val jsonFormats: Formats = DefaultFormats

  protected val applicationDescription: String = "SchemaApi"
  override protected val applicationName: Option[String] = Some("schema")

  before() {
    contentType = formats("json")
    response.headers += ("Access-Control-Allow-Origin" -> "*")
  }



  val findOperation = (apiOperation[Any]("find")
      summary "Get model schemata for data objects returned by this API."
      parameters(
        queryParam[String]("model").description("").optional)
  )

  get("/",operation(findOperation)) {
    val model = params.getAs[String]("model")
    println("model: " + model)
  }




  val websocketHelpOperation = (apiOperation[Any]("websocketHelp")
      summary "Returns help text & subject list for websocket usage."
      parameters(
        )
  )

  get("/websocketHelp",operation(websocketHelpOperation)) {
    }

}
